packages/model-viewer/src/features/scene-graph/image.ts

```diff
@@ -41,30 +41,34 @@ export class Image extends ThreeDOMElement implements ImageInterface {
     return this[$correlatedObjects]?.values().next().value as ThreeTexture;
   }
 
-  constructor(
-      onUpdate: () => void, texture: ThreeTexture|null,
-      gltfImage: GLTFImage|null) {
-    gltfImage = gltfImage ?? {
-      name: (texture && texture.image && texture.image.src) ?
+  get[$threeTextures](): Set<ThreeTexture> {
+    return this[$correlatedObjects] as Set<ThreeTexture>;
+  }
+
+  constructor(onUpdate: () => void, texture: ThreeTexture|null) {
```

Comment:
I know this predates you, but do we know when `|null` happens? There's an assert above that seems to imply `null` would be an error. Doesn't have to happen in this PR, but it would be great to clean this up if we can.

Reply:
Yes, or at least it should work the same way as https://github.com/google/model-viewer/blob/8546f41450af57ee37fe7c00e3ef210138f6e3a0/packages/model-viewer/src/features/scene-graph/sampler.ts#L80C2-L100C56

Maybe another refactor where we remove `|null` from all ThreeDOMElements (Sampler, Image, Texture, etc...) is something worth looking into.

Reply:
Yes, I think that would be ideal.